src/main/java/net/minecraftforge/common/util/RotationHelper.java

```diff
@@ -0,0 +1,170 @@
+package net.minecraftforge.common.util;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockAnvil;
+import net.minecraft.block.BlockBed;
+import net.minecraft.block.BlockButton;
+import net.minecraft.block.BlockChest;
+import net.minecraft.block.BlockCocoa;
+import net.minecraft.block.BlockDispenser;
+import net.minecraft.block.BlockDoor;
+import net.minecraft.block.BlockEndPortalFrame;
+import net.minecraft.block.BlockEnderChest;
+import net.minecraft.block.BlockFenceGate;
+import net.minecraft.block.BlockFurnace;
+import net.minecraft.block.BlockHay;
+import net.minecraft.block.BlockHopper;
+import net.minecraft.block.BlockHugeMushroom;
+import net.minecraft.block.BlockLadder;
+import net.minecraft.block.BlockLever;
+import net.minecraft.block.BlockLog;
+import net.minecraft.block.BlockNewLog;
+import net.minecraft.block.BlockPistonBase;
+import net.minecraft.block.BlockPistonExtension;
+import net.minecraft.block.BlockPumpkin;
+import net.minecraft.block.BlockQuartz;
+import net.minecraft.block.BlockStandingSign;
+import net.minecraft.block.BlockLog.EnumAxis;
+import net.minecraft.block.BlockRail;
+import net.minecraft.block.BlockRailDetector;
+import net.minecraft.block.BlockRailPowered;
+import net.minecraft.block.BlockRedstoneComparator;
+import net.minecraft.block.BlockRedstoneRepeater;
+import net.minecraft.block.BlockSkull;
+import net.minecraft.block.BlockStairs;
+import net.minecraft.block.BlockTorch;
+import net.minecraft.block.BlockTrapDoor;
+import net.minecraft.block.BlockTripWireHook;
+import net.minecraft.block.BlockVine;
+import net.minecraft.block.BlockWallSign;
+import net.minecraft.block.properties.IProperty;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.BlockPos;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.world.World;
+
+public class RotationHelper
+{
+
+    static public EnumAxis fromQuartz( BlockQuartz.EnumType type )
+    {
+        switch(type)
+        {
+            case LINES_X:
+                return EnumAxis.X;
+            case LINES_Y:
+                return EnumAxis.Y;
+            case LINES_Z:
+                return EnumAxis.Z;
+            default:
+                return EnumAxis.NONE;
+        }
+    }
+    
+    static public boolean canRotateBlock( Block block )
+    {
+        return block instanceof BlockQuartz ||
+                block instanceof BlockLog ||
```

Comment:
If you wanted a face sensitive implementation:
`facing`: default `EnumFacing`, levers `BlockLever.EnumOrientation`
`axis`: default `EnumFacing.Axis`, logs `BlockLog.EnumAxis`
`variant`: quartz block `BlockQuartz.EnumType`

[That's all](https://gist.github.com/7d46d2512ffe2db042ef) I've found myself. The implementation is a bit different because I do a few extra things (`x != null / logger.error` and forced checks are not needed). Some of the stuff (`FACINGS`, `AXISES`) should be part of `EnumFacing` but aren't.